draft-wang-ppm-differential-privacy.md

```diff
@@ -456,9 +523,105 @@ class DpMechanism:
 
 > TODO: Specify a Gaussian sampler
 
-## Randomized Response {#rr}
+## Binary Randomized Response {#rr}
+
+The binary randomized response mechanism specified in Appendix C.1 of
+{{MJTB+22}} is initialized with a parameter `EPSILON_0`, and takes in a single
+bit `x`. The bit is flipped to `1 - x` with probability
+`1 / (exp(EPSILON_0) + 1)`. For example, if `EPSILON_0` is configured to be 3,
+and the input to binary randomized response is a 0, the bit will be flipped to
+be 1 with probability `1 / (exp(3) + 1)`, otherwise, it will stay as a 0.
+
+Under OC trust model, by applying binary randomized response with `EPSILON_0`
+parameter to its measurement, the Client gets `EPSILON_0`-DP in deletion-DP
+model (Definition II.4 of {{EFMR+20}}, and Definition C.1 of {{MJTB+22}}).
+A formal definition of deletion-DP is elaborated in {{rr-deletion-dp}}.
+
+We generalize binary randomized response mechanism by applying it independently
+at all coordinates of a Client's bit vector. Under OAMC trust model, it is
+proven in Appendix C.1.3 of {{MJTB+22}} that we get good `(EPSILON, DELTA)`-DP
+in the replacement-DP model, by aggregating a batch of noisy Client
+measurements, each of which is a bit vector with exactly one bit set, and is
+noised with binary randomized response at all coordinates. The final aggregate
+result needs to be "debiased" due to the noise added by the Clients, and is
+expressed as a vector of floats, because of floating point arithmetic.
+
+Since the noise generated by each Client at each coordinate is independent, and
+as the number of Clients `n` grows, the noise distribution at each coordinate
+approximates a Gaussian distribution, with standard deviation
+`sqrt(n * exp(EPSILON_0) / (exp(EPSILON_0) - 1)^2)`, as proved by Theorem C.2 of
+{{MJTB+22}}.
+
+### `EPSILON_0`-DP in Deletion-DP {#rr-deletion-dp}
```

Comment:
This section is well-scoped, and I don't want to change anything for now.

In the medium term, I think we should spend time thinking through how this notion of DP relates to the notion of DP we've already defined.

Long term, I think our goal for PPM should be to have a single (or as few as possible) notions of DP. That way we have a standard methodology (or methodologies) for evaluating DP mechanisms and policies. The fact that there are so many DP notions in the literature that may or may not be comparable is a huge problem for standardization.

Relatedly, I believe there is an important research question here: for a given application (DAP) systematize the DP notions in the literature that apply such that we can compare them to one another (at least qualitatively, but ideally quantitatively). Presumably there are trade-offs to be made, depending on which notion of DP you pick; what we want at the end of the day is the right DP notion that allows us to make sense of what to do for DAP.

Reply:
Agreed. Let's continue discussing in issue.